File: src/test/java/info/dreamcoder/columns/DBStringTest.java
package info.dreamcoder.columns;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.*;

@DisplayName("测试字符串类型的列")
class DBStringTest {

    @Test
    @DisplayName("能正确的创建字符串的类型，默认是255")
    void shouldCreateStringColumn() {
        DBString col = new DBString("string_column");
        assertThat(col.toSql()).isEqualTo("string_column varchar(255)");
    }

    @Test
    @DisplayName("可以设置字符串类型的列的长度")
    void shouldSetColumnLength() {
        DBString col = new DBString("string_column").limit(150);
        assertThat(col.toSql()).isEqualTo("string_column varchar(150)");
    }
}

File: src/test/kotlin/columns/DbStringTest.kt
package columns

import io.kotest.core.spec.DisplayName
import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe

@DisplayName("测试类 DbString")
class DbStringTest : DescribeSpec({

    describe("toSql方法") {
        it("默认生成250 varchar类型的列") {
            val column = DbString("test_column")
            column.toSql() shouldBe "test_column varchar(250)"
        }
    }

})
